### PR TITLE
Shutdown node when component startup fails

### PR DESCRIPTION
**Pull request by mavam**

### CODE REVIEW

**Review by lava — APPROVED** [review]
Looks good to me. Ideally we'd have a test for this, but of course shutdown-related tests are notoriously hard to write. Maybe we could somehow add a corrupted index/archive state to the repository and try to restore it in one of the integration tests?

File: libvast/src/system/node.cpp
Comment (mavam) [reply libvast/src/system/node.cpp:395]: How do you feel about *"The node monitors all components in the registry"*? I'm trying to make it clear who is the monitoring entity, because the component registry is just a container.

**Comment by mavam** [comment]
> Maybe we could somehow add a corrupted index/archive state to the repository and try to restore it in one of the integration tests?

Good idea. Let's hold off with that until you've merged the flatbuffer changes. I'll create a follow-up story.

**Pull request merged**
